frontend/src/metabase/visualizations/lib/settings/graph.js

```diff
@@ -152,13 +153,18 @@ export const GRAPH_DATA_SETTINGS = {
       const options = data.cols
         .filter(vizSettings["graph._metric_filter"])
         .map(getOptionFromColumn);
+
+      const addedMetricsCount = vizSettings["graph.metrics"].length;
+      const supportedMetrics = getMaxMetricsSupported(card.display);
+
+      const canAddAnother =
+        addedMetricsCount < supportedMetrics &&
+        options.length > value.length &&
+        vizSettings["graph.dimensions"].length < 2;
```

Comment:
Mind improving the conditionals readability here?

`addedMetricsCount < supportedMetrics` is very straightforward, but the following conditions can be confusing, I guess. (btw, `supportedMetrics` sounds like a list of metrics to me, and not the number. Should we maybe call it like `maxMetricsCount` or `maxMetricsCountForDisplay`?)

`options.length > value.length` can be something like `hasMetricsToAdd` (if I understand the meaning behind it correctly)

`vizSettings["graph.dimensions"].length < 2` is a bit of a mystery for me TBH. Does `2` mean X and Y axes? Fixing the magic number and giving a meaningful name to the comparison can make things easier, I believe

Reply:
Thanks for pointing it out. I've made it more readable now